sql-plugin/src/main/301until320-nondb/scala/com/nvidia/spark/rapids/shims/v2/Spark30XShims.scala

```diff
@@ -142,4 +142,14 @@ trait Spark30XShims extends SparkShims {
   override def shouldFallbackOnAnsiTimestamp(): Boolean = false
 
   override def getLegacyStatisticalAggregate(): Boolean = true
+
+  override def supportsColumnarAdaptivePlans: Boolean = false
+
+  override def columnarAdaptivePlan(a: AdaptiveSparkPlanExec, goal: CoalesceSizeGoal): SparkPlan = {
+    // When the input is an adaptive plan we do not get to see the GPU version until
+    // the plan is executed and sometimes the plan will have a GpuColumnarToRowExec as the
+    // final operator and we can bypass this to keep the data columnar by inserting
+    // the [[GpuAvoidAdaptiveTransitionToRow]] operator here
+    AvoidAdaptiveTransitionToRow(GpuRowToColumnarExec(a, goal))
```

Comment:
We are not checking for preProcessing tags here and I'm not sure if we should be in this case.

Reply:
Do we need to update the Databricks shims with the same change?  I see AvoidAdaptiveTransitionToRow class declared in the 301until320 v2 shim, but it seems to only be called in the non-db cases.  We don't support AQE on Databricks, but if that problem gets solved it would be nice to not be surprised with all the things we could have updated for AQE in Databricks shims but did not.